src/TodoList.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import TodoItem from './TodoItem';
+
+class TodoList extends React.PureComponent {
+  state = {
+    todoList: this.props.todoList,
+    // allVisible: true,
+    // activeVisible: false,
+    // completedVisible: false,
+  }
+
+  addNewTodo = (newTodo) => {
```

Comment:
all actions with the initial array of todos should be done in the main parent component (e.g. `App.js` or `TodoApp.js`) since you need this info in other components except TodoList